### PR TITLE
Add Sparse embedding support

### PR DESCRIPTION
**Pull request by kalyc**
### Summary
Add minimal test for testing sparse embedding operator support

### Related Issues
[Missing Sparse operator support](https://github.com/awslabs/keras-apache-mxnet/issues/18)

### PR Overview
- [y] This PR requires new unit tests [y/n] (make sure tests are included)
- [n] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [y] This PR is backwards compatible [y/n]
- [n] This PR changes the current API [y/n]

### CODE REVIEW

File: tests/keras/backend/mxnet_sparse_test.py [hunk 1]
```diff
@@ -104,6 +104,29 @@ def test_sparse_dot(self):
...
+        dense_matrix = sparse_matrix.toarray()
+        test_dense_data = K.variable(dense_matrix)
+
+        dense_weight = sparse_weight.toarray()
+        test_dense_weight = K.variable(dense_weight)
+
+        k_S = K.embedding(test_sparse_data, test_sparse_weight, 4, 5)
+        k_D = K.embedding(test_dense_data, test_dense_weight, 4, 5)
+
+        assert k_S.shape == k_D.shape
```
Comment (sandeep-krishnamurthy) [inline tests/keras/backend/mxnet_sparse_test.py]: value verification?

Comment (kalyc) [reply tests/keras/backend/mxnet_sparse_test.py]: Added, also see related issue about using contrib API - https://github.com/apache/incubator-mxnet/issues/12465

File: keras/backend/mxnet_backend.py [hunk 2]
```diff
@@ -1202,13 +1202,18 @@ def gather(reference, indices):
 
 
 @keras_mxnet_symbol
-def embedding(data, weight, input_dim, output_dim):
+def embedding(data, weight, input_dim, output_dim, sparse_grad=False):
```
Comment (roywei) [inline keras/backend/mxnet_backend.py]: K.embedding() is called in keras.layers.Embedding class for mxnet backend. Please also update the function signature there, and test in a end to end example that use Embedding layer. (e.g. examples using imdb dataset)

File: tests/keras/backend/mxnet_sparse_test.py [hunk 3]
```diff
@@ -104,6 +105,43 @@ def test_sparse_dot(self):
...
 
+    def _forward_pass(self, x):
+        bind_values = K.dfs_get_bind_values(x)
+        executor = x.symbol.simple_bind(mx.cpu(), grad_req='null')
+        for v in executor.arg_dict:
+            bind_values[v].copyto(executor.arg_dict[v])
+        outputs = executor.forward(is_train=K.learning_phase())
+        return outputs
+
+    def test_sparse_embedding(self):
```
Comment (roywei) [inline tests/keras/backend/mxnet_sparse_test.py]: Could you add a test similar to tests/keras/layers/embedding_test.py? use  layer_test to test Embedding layer class

File: keras/backend/mxnet_backend.py (hunk 2 above)
Comment (kalyc) [reply keras/backend/mxnet_backend.py]: We are passing default value as `False` for `sparse_grad` in the API signature so making a change there is not necessary

**Comment by sandeep-krishnamurthy** [comment]
@kalyc - Can we move ahead with this, as we discussed to use mxnet --preview package?

File: keras/layers/embeddings.py [hunk 4]
```diff
@@ -78,14 +78,14 @@ def __init__(self, input_dim, output_dim,
                  embeddings_constraint=None,
                  mask_zero=False,
                  input_length=None,
+                 sparse_grad=False,
```
Comment (roywei) [inline keras/layers/embeddings.py]: Add this in doc string explaining the usage and note it's only for mxnet backend

File: tests/keras/backend/mxnet_sparse_test.py [hunk 5]
```diff
@@ -160,6 +160,5 @@ def test_sparse_concat_axis_non_zero(self):
         assert k_s_d.shape == k_d.shape
         assert_allclose(k_s_d, k_d, atol=1e-05)
 
-
```
Comment (roywei) [inline tests/keras/backend/mxnet_sparse_test.py]: Please add sparse test here, add a case where sparse_grad is true:  https://github.com/awslabs/keras-apache-mxnet/blob/master/tests/keras/layers/embeddings_test.py

File: keras/layers/embeddings.py [hunk 6]
```diff
@@ -140,7 +141,10 @@ def call(self, inputs):
         # K.gather is not working with Embedding layer using MXNet backend
         # Refer to this issue: https://github.com/awslabs/keras-apache-mxnet/issues/63
         if K.backend() == "mxnet":
-            out = K.embedding(inputs, self.embeddings, self.input_dim, self.output_dim)
+            if self.sparse_grad:
+                out = K.embedding(inputs, self.embeddings, self.input_dim, self.output_dim, sparse_grad=True)
```
[the remark this answers is not in the document]
Comment (kalyc) [reply keras/layers/embeddings.py]: the condition will pass only when self.sparse_grad=True, updated the function call for more clarity

[the remark this answers is not in the document]
Comment (roywei) [reply keras/layers/embeddings.py]: can we simplify to not use if else:   
`out = K.embedding(inputs, self.embeddings, self.input_dim, self.output_dim, sparse_grad=self.sparse_grad)`

**Pull request merged**